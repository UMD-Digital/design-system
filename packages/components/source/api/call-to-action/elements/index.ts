import { Composite } from '@universityofmaryland/web-elements-library';
import { Tokens } from '@universityofmaryland/variables';
import { Styles, Markup } from 'utilities';
import { ELEMENTS, SLOTS, VARIABLES, REFERENCES } from '../globals';
import { UMDCallToActionElement } from '../index';

const { Colors, FontSize, Spacing } = Tokens;
const { CallToAction } = Composite;
const { Node } = Markup.create;

const { PLAIN_TEXT } = SLOTS;
const { CTA_CONTAINER } = ELEMENTS;
const { ATTRIBUTE_PLAIN_TEXT, ATTRIBUTE_TYPE } = VARIABLES;
const { IS_PRIMARY, IS_SECONDARY, IS_PLAIN_TEXT } = REFERENCES;

const CTA_PLAIN_TEXT_SLOT = 'umd-call-to-action-plain-text-slot';

// prettier-ignore
const OverwritePrimaryLayoutStyles = `
  ${IS_PRIMARY}${IS_PLAIN_TEXT} {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  ${IS_PRIMARY} .${CTA_PLAIN_TEXT_SLOT} {
    margin-top: ${Spacing.min};
  }
`

// prettier-ignore
const OverwriteSecondaryLayoutStyles = `
  ${IS_SECONDARY}${IS_PLAIN_TEXT} {
    display: flex;
    flex-direction: column;
  }

  ${IS_SECONDARY} .${CTA_PLAIN_TEXT_SLOT} {
    padding-left: 20px;
    margin-top: ${Spacing.min};
  }
`

// prettier-ignore
const PlainTextSlotStyles = `
  .${CTA_PLAIN_TEXT_SLOT} {
    text-decoration: underline;
    display: inline-block;
    font-size: ${FontSize.min};
    transition: color .5s;
    position: relative;
  }

  .${CTA_PLAIN_TEXT_SLOT}:hover,
  .${CTA_PLAIN_TEXT_SLOT}:focus {
    color: ${Colors.redDark};
  }
`;

// prettier-ignore
export const ComponentStyles = `
  :host {
    display: block;
  }

  ${Styles.reset}
  ${CallToAction.Styles}
  ${PlainTextSlotStyles}
  ${OverwritePrimaryLayoutStyles}
  ${OverwriteSecondaryLayoutStyles}
`;

const CreateCtaElement = ({ element }: { element: UMDCallToActionElement }) => {
  const type = element._type;
  const size = element._size;
  const isThemeDark = element._isThemeDark;
  const isThemeGold = element._isThemeGold;
  const styleProps = element._styleProps;

  const linkElement = element.querySelector(`a`);
  const buttonElement = element.querySelector(`button`);
  const styleObj = { type, size, isThemeDark, isThemeGold, styleProps };

  if (linkElement) {
    const cta = linkElement.cloneNode(true) as HTMLAnchorElement;
    return CallToAction.CreateElement({ cta, ...styleObj });
  }
  if (buttonElement) {
    const cta = buttonElement.cloneNode(true) as HTMLButtonElement;
    return CallToAction.CreateElement({ cta, ...styleObj });
  }

  return CallToAction.CreateElement({ type });
};

const CreatePlainText = ({ element }: { element: UMDCallToActionElement }) => {
  const linkElements = Array.from(element.querySelectorAll(`a`));
  return linkElements.filter(
    (link) => link.getAttribute('slot') === PLAIN_TEXT,
  );
};

export const CreateShadowDom = ({
  element,
}: {
  element: UMDCallToActionElement;
}) => {
  const type = element._type;
  const container = document.createElement('div');
  const ctaElement = CreateCtaElement({ element });
  const plainTextElement = CreatePlainText({ element });
  const hasPlainText = plainTextElement.length > 0;

  container.classList.add(CTA_CONTAINER);
  container.appendChild(ctaElement);
  container.setAttribute(ATTRIBUTE_TYPE, type);

  if (hasPlainText) {
    const plainTextSlot = Node.slot({ type: PLAIN_TEXT });
    plainTextSlot.classList.add(CTA_PLAIN_TEXT_SLOT);
    container.setAttribute(ATTRIBUTE_PLAIN_TEXT, 'true');
    container.appendChild(plainTextSlot);
  }

  return container;
};
