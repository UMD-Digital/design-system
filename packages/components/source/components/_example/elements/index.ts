import { Styles } from 'utilities';

export const ComponentStyles = `
  :host {
    display: block;
  }

  ${Styles.ResetString}
`;

export const CreateShadowDom = ({ element }: { element: HTMLElement }) => {
  const container = document.createElement('div');
  return container;
};
