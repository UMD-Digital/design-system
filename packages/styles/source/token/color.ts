/**
 * @module token/color
 * Provides color tokens for the design system.
 */

export type ColorScale = typeof colorScale;
export type BaseColors = typeof baseColors;

/**
 * Color scale with various shades for different colors.
 * @returns {ColorScale} Color scale object with various color categories and shades.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.colorScale.gray[500]
 * ```
 * @since 1.1.0
 */
export const colorScale = {
  gray: {
    50: '#FAFAFA',
    100: '#F1F1F1',
    200: '#E6E6E6',
    400: '#7F7F7F',
    500: '#757575',
    700: '#454545',
    900: '#242424',
  },
  red: {
    500: '#E21833',
    900: '#A90007',
  },
  blue: {
    500: '#2F7EDA',
  },
  green: {
    500: '#008000',
  },
  gold: {
    500: '#FFD200',
  },
} as const;

/**
 * Base colors for black and white.
 * @returns {BaseColors} Base colors object with white and black.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.baseColors.white
 * ```
 * @since 1.1.0
 */
export const baseColors = {
  white: '#FFFFFF',
  black: '#000000',
} as const;

/**
 * Default color tokens for the design system.
 * @returns {object} Color tokens with semantic naming.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.color.red
 * ```
 * @since 1.1.0
 */
export default {
  red: colorScale.red[500],
  blue: colorScale.blue[500],
  green: colorScale.green[500],
  gold: colorScale.gold[500],
  white: baseColors.white,
  black: baseColors.black,
  redDark: colorScale.red[900],
  gray: {
    darker: colorScale.gray[900],
    dark: colorScale.gray[700],
    mediumAA: colorScale.gray[500],
    medium: colorScale.gray[400],
    light: colorScale.gray[200],
    lighter: colorScale.gray[100],
    lightest: colorScale.gray[50],
  },
};
