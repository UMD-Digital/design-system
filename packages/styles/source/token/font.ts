/**
 * @module token/font
 * Provides font-related design tokens including sizes, weights, and font families.
 */

/**
 * Font size tokens.
 * @type {Object<string, string>}
 * @property {string} min - 12px font size
 * @property {string} sm - 14px font size
 * @property {string} base - 16px font size (default)
 * @property {string} lg - 18px font size
 * @property {string} xl - 20px font size
 * @property {string} 2xl - 22px font size
 * @property {string} 3xl - 24px font size
 * @property {string} 4xl - 32px font size
 * @property {string} 5xl - 48px font size
 * @property {string} 6xl - 56px font size
 * @property {string} 7xl - 64px font size
 * @property {string} 8xl - 72px font size
 * @property {string} 9xl - 80px font size
 * @property {string} 10xl - 96px font size
 * @property {string} max - 120px font size
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font.size.base // '16px'
 * ```
 * @since 1.8.0
 */
export const size = {
  min: '12px',
  sm: '14px',
  base: '16px',
  lg: '18px',
  xl: '20px',
  '2xl': '22px',
  '3xl': '24px',
  '4xl': '32px',
  '5xl': '48px',
  '6xl': '56px',
  '7xl': '64px',
  '8xl': '72px',
  '9xl': '80px',
  '10xl': '96px',
  max: '120px',
};

/**
 * Font weight tokens.
 * @type {Object<string, string>}
 * @property {string} thin - 100 weight
 * @property {string} extraLight - 200 weight
 * @property {string} light - 300 weight
 * @property {string} normal - 400 weight (default)
 * @property {string} medium - 500 weight
 * @property {string} semiBold - 600 weight
 * @property {string} bold - 700 weight
 * @property {string} extraBold - 800 weight
 * @property {string} black - 900 weight
 * @property {string} extraBlack - 950 weight
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font.weight.bold // '700'
 * ```
 * @since 1.8.0
 */
export const weight = {
  thin: '100',
  extraLight: '200',
  light: '300',
  normal: '400',
  medium: '500',
  semiBold: '600',
  bold: '700',
  extraBold: '800',
  black: '900',
  extraBlack: '950',
};

/**
 * Font family tokens.
 * @type {Object<string, string>}
 * @property {string} campaign - Barlow Condensed font stack for campaign/marketing text
 * @property {string} sans - Interstate font stack for primary sans-serif text
 * @property {string} serif - Crimson Pro font stack for serif text
 * @property {string} mono - Monospace font stack for code/technical text
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font.family.sans // "'Interstate', Helvetica, Arial, Verdana, sans-serif"
 * ```
 * @since 1.8.0
 */
export const family = {
  campaign: "'Barlow Condensed', Arial Narrow, sans-serif",
  sans: "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  serif: "'Crimson Pro', Georgia, serif",
  mono: 'monospace',
};
