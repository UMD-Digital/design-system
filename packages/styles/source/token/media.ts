/**
 * @module token/media
 * Provides responsive design tokens for breakpoints and media queries.
 */

/**
 * Numeric breakpoint values in pixels.
 * @type {Object<string, {min: number, max?: number}>}
 * @property {Object} small - Small screen breakpoint (320px-479px)
 * @property {Object} medium - Medium screen breakpoint (480px-649px)
 * @property {Object} large - Large screen breakpoint (650px-767px) 
 * @property {Object} tablet - Tablet breakpoint (768px-1023px)
 * @property {Object} desktop - Desktop breakpoint (1024px-1199px)
 * @property {Object} highDef - High-definition screen breakpoint (1200px-1499px)
 * @property {Object} maximum - Maximum screen size breakpoint (1500px+)
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.media.breakpointValues.desktop.min // 1024
 * ```
 * @since 1.8.0
 */
export const breakpointValues = {
  small: { min: 320, max: 479 },
  medium: { min: 480, max: 649 },
  large: { min: 650, max: 767 },
  tablet: { min: 768, max: 1023 },
  desktop: { min: 1024, max: 1199 },
  highDef: { min: 1200, max: 1499 },
  maximum: { min: 1500 },
};

/**
 * Breakpoint values with 'px' units for CSS usage.
 * @type {Object<string, {min: string, max?: string}>}
 * @property {Object} small - Small screen breakpoint strings ("320px"-"479px")
 * @property {Object} medium - Medium screen breakpoint strings ("480px"-"649px")
 * @property {Object} large - Large screen breakpoint strings ("650px"-"767px")
 * @property {Object} tablet - Tablet breakpoint strings ("768px"-"1023px")
 * @property {Object} desktop - Desktop breakpoint strings ("1024px"-"1199px")
 * @property {Object} highDef - High-definition screen breakpoint strings ("1200px"-"1499px")
 * @property {Object} maximum - Maximum screen size breakpoint string ("1500px"+)
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.media.breakpoints.desktop.min // "1024px"
 * ```
 * @since 1.8.0
 */
export const breakpoints = {
  small: {
    min: `${breakpointValues.small.min}px`,
    max: `${breakpointValues.small.max}px`,
  },
  medium: {
    min: `${breakpointValues.medium.min}px`,
    max: `${breakpointValues.medium.max}px`,
  },
  large: {
    min: `${breakpointValues.large.min}px`,
    max: `${breakpointValues.large.max}px`,
  },
  tablet: {
    min: `${breakpointValues.tablet.min}px`,
    max: `${breakpointValues.tablet.max}px`,
  },
  desktop: {
    min: `${breakpointValues.desktop.min}px`,
    max: `${breakpointValues.desktop.max}px`,
  },
  highDef: {
    min: `${breakpointValues.highDef.min}px`,
    max: `${breakpointValues.highDef.max}px`,
  },
  maximum: { min: `${breakpointValues.maximum.min}px` },
};

/**
 * Complete media query strings ready for use in CSS.
 * @type {Object<string, {min: string, max?: string}>}
 * @property {Object} small - Small screen media queries
 * @property {Object} medium - Medium screen media queries
 * @property {Object} large - Large screen media queries
 * @property {Object} tablet - Tablet media queries
 * @property {Object} desktop - Desktop media queries
 * @property {Object} highDef - High-definition screen media queries
 * @property {Object} maximum - Maximum screen size media query
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * // Use in JSS/CSS-in-JS
 * const styles = {
 *   [`@media (${Styles.token.media.queries.desktop.min})`]: {
 *     // Desktop-specific styles
 *   }
 * };
 * ```
 * @since 1.8.0
 */
export const queries = {
  small: {
    min: `min-width: ${breakpoints.small.min}`,
    max: `max-width: ${breakpoints.small.max}`,
  },
  medium: {
    min: `min-width: ${breakpoints.medium.min}`,
    max: `max-width: ${breakpoints.medium.max}`,
  },
  large: {
    min: `min-width: ${breakpoints.large.min}`,
    max: `max-width: ${breakpoints.large.max}`,
  },
  tablet: {
    min: `min-width: ${breakpoints.tablet.min}`,
    max: `max-width: ${breakpoints.tablet.max}`,
  },
  desktop: {
    min: `min-width: ${breakpoints.desktop.min}`,
    max: `max-width: ${breakpoints.desktop.max}`,
  },
  highDef: {
    min: `min-width: ${breakpoints.highDef.min}`,
    max: `max-width: ${breakpoints.highDef.max}`,
  },
  maximum: { min: `min-width: ${breakpoints.maximum.min}` },
};
