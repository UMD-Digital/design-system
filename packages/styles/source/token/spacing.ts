/**
 * @module token/spacing
 * Provides spacing tokens for the design system.
 */

/**
 * Spacing scale with values in pixels and their corresponding rem equivalents.
 * @returns {object} Spacing scale with numeric keys and pixel values.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.spacingScale[4] // '16px'
 * ```
 * @since 1.1.0
 */
export const spacingScale = {
  0: '0',
  1: '4px', // 0.25rem
  2: '8px', // 0.5rem  (min)
  3: '12px', // 0.75rem (xs)
  4: '16px', // 1rem    (sm)
  6: '24px', // 1.5rem  (md)
  8: '32px', // 2rem    (lg)
  10: '40px', // 2.5rem  (xl)
  12: '48px', // 3rem    (2xl)
  14: '56px', // 3.5rem  (3xl)
  16: '64px', // 4rem    (4xl)
  18: '72px', // 4.5rem  (5xl)
  20: '80px', // 5rem    (6xl)
  24: '96px', // 6rem    (7xl)
  26: '104px', // 6.5rem  (8xl)
  30: '120px', // 7.5rem  (max)
} as const;

/**
 * Maximum width constraints for layout containers.
 * @returns {object} Maximum width values with semantic naming.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.maxWidth.normal // '1384px'
 * ```
 * @since 1.1.0
 */
export const maxWidth = {
  max: '1728px',
  large: '1552px',
  normal: '1384px',
  small: '1200px',
  smallest: '992px',
};

/**
 * Default spacing tokens with semantic naming.
 * @returns {object} Spacing tokens with semantic names and corresponding pixel values.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.spacing.md // '24px'
 * ```
 * @since 1.1.0
 */
export default {
  min: '8px',
  xs: '12px',
  sm: '16px',
  md: '24px',
  lg: '32px',
  xl: '40px',
  '2xl': '48px',
  '3xl': '56px',
  '4xl': '64px',
  '5xl': '72px',
  '6xl': '80px',
  '7xl': '96px',
  '8xl': '104px',
  max: '120px',
  maxWidth: {
    ...maxWidth,
  },
};
