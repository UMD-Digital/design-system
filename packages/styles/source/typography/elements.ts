import { colors, font, media } from '../tokens';
import { create } from '../utilities';

const breakpointLarge = media.queries.large.min;
const breakpointDesktop = media.queries.desktop.min;

export const eyebrow = {
  fontWeight: 700,
  color: colors.black,
  fontSize: font.size.min,
  lineHeight: 1,
  textTransform: `uppercase`,
  letterSpacing: `.05em`,
};

export const labelMedium = {
  fontFamily: font.family['sans'],
  fontSize: font.size['sm'],
  lineHeight: `1.42em`,

  [`@media (${breakpointLarge})`]: {
    fontSize: `calc(${font.size['sm']} + 0.16vw)`,
  },

  [`@media (${breakpointDesktop})`]: {
    fontSize: font.size['base'],
    letterSpacing: '0.16em',
    lineHeight: `1.5em`,
  },
};

export const labelSmall = {
  fontFamily: font.family['sans'],
  fontSize: font.size['sm'],
  letterSpacing: '0.16em',
  lineHeight: `1.42em`,
};

export const interativeMedium = {
  fontFamily: font.family['sans'],
  fontSize: font.size['base'],
  lineHeight: `1.125em`,

  [`@media (${breakpointLarge})`]: {
    fontSize: `calc(${font.size['base']} + 0.16vw)`,
  },

  [`@media (${breakpointDesktop})`]: {
    fontSize: font.size['lg'],
    fontWeight: font.weight['bold'],
    letterSpacing: '-0.01em',
    lineHeight: `1.11em`,
  },
};

export const interativeSmall = {
  fontFamily: font.family['sans'],
  fontSize: font.size['base'],
  fontWeight: font.weight['bold'],
  letterSpacing: '-0.01em',
  lineHeight: `1.125.em`,
};

export const fonts = {
  eyebrow: create.jssObject({
    className: 'umd-eyebrow',
    ...eyebrow,
  }),

  labelMedium: create.jssObject({
    className: 'umd-label-sans-medium',
    ...labelMedium,
  }),

  labelSmall: create.jssObject({
    className: 'umd-label-sans-small',
    ...labelSmall,
  }),

  medium: create.jssObject({
    className: 'umd-interactive-sans-medium',
    ...interativeMedium,
  }),

  small: create.jssObject({
    className: 'umd-interactive-sans-small',
    ...interativeSmall,
  }),
};
