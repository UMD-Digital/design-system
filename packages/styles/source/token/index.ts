/**
 * @module token
 * Design tokens and system variables.
 */

import color from './color';
import spacing from './spacing';

/**
 * Media query breakpoints and utilities.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.media
 * ```
 * @since 1.1.0
 */
export * as media from './media';

/**
 * Typography and font tokens.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.font
 * ```
 * @since 1.1.0
 */
export * as font from './font';

/**
 * Color system tokens.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.color
 * ```
 * @since 1.1.0
 */
export { color };

/**
 * Spacing system tokens.
 * @example
 * ```typescript
 * import * as Styles from '@universityofmaryland/web-styles-library';
 * Styles.token.spacing
 * ```
 * @since 1.1.0
 */
export { spacing };
