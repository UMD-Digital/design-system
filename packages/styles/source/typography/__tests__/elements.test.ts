import * as elements from '../elements';
import { font } from '../../token';

describe('typography/elements', () => {
  it('should match snapshot', () => {
    expect(elements).toMatchSnapshot();
  });

  describe('element typography objects', () => {
    it('should contain typography styles for different UI elements', () => {
      // Test for common element typography exports
      expect(elements.eyebrow).toBeDefined();
      expect(elements.labelMedium).toBeDefined();
      expect(elements.labelSmall).toBeDefined();
      expect(elements.interativeMedium).toBeDefined();
      expect(elements.interativeSmall).toBeDefined();
    });

    it('should use appropriate font styles for elements', () => {
      // Elements should have appropriate font family
      expect(elements.labelMedium.fontFamily).toBe(font.family.sans);
      expect(elements.labelSmall.fontFamily).toBe(font.family.sans);
      expect(elements.interativeMedium.fontFamily).toBe(font.family.sans);
      expect(elements.interativeSmall.fontFamily).toBe(font.family.sans);
    });

    it('should match snapshots for element typography', () => {
      expect(elements.eyebrow).toMatchSnapshot('eyebrow');
      expect(elements.labelMedium).toMatchSnapshot('labelMedium');
      expect(elements.labelSmall).toMatchSnapshot('labelSmall');
      expect(elements.interativeMedium).toMatchSnapshot('interactiveMedium');
      expect(elements.interativeSmall).toMatchSnapshot('interactiveSmall');
    });
  });

  describe('fonts objects', () => {
    it('should have correctly named font objects with classNames', () => {
      // Check if the fonts object contains class definitions for the elements
      expect(elements.fonts.eyebrow.className).toBe('umd-eyebrow');
      expect(elements.fonts.labelMedium.className).toBe('umd-label-sans-medium');
      expect(elements.fonts.labelSmall.className).toBe('umd-label-sans-small');
      expect(elements.fonts.medium.className).toBe('umd-interactive-sans-medium');
      expect(elements.fonts.small.className).toBe('umd-interactive-sans-small');
    });

    it('should match snapshot for all fonts objects', () => {
      expect(elements.fonts).toMatchSnapshot();
    });
  });
});