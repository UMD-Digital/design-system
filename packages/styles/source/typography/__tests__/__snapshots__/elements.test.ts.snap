// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typography/elements element typography objects should match snapshots for element typography: eyebrow 1`] = `
{
  "color": "#000000",
  "fontSize": "12px",
  "fontWeight": 700,
  "letterSpacing": ".05em",
  "lineHeight": 1,
  "textTransform": "uppercase",
}
`;

exports[`typography/elements element typography objects should match snapshots for element typography: interactiveMedium 1`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "18px",
    "fontWeight": "700",
    "letterSpacing": "-0.01em",
    "lineHeight": "1.11em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(16px + 0.16vw)",
  },
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "16px",
  "lineHeight": "1.125em",
}
`;

exports[`typography/elements element typography objects should match snapshots for element typography: interactiveSmall 1`] = `
{
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "16px",
  "fontWeight": "700",
  "letterSpacing": "-0.01em",
  "lineHeight": "1.125em",
}
`;

exports[`typography/elements element typography objects should match snapshots for element typography: labelMedium 1`] = `
{
  "@media (min-width: 1024px)": {
    "fontSize": "16px",
    "letterSpacing": "0.16em",
    "lineHeight": "1.5em",
  },
  "@media (min-width: 650px)": {
    "fontSize": "calc(14px + 0.16vw)",
  },
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "14px",
  "lineHeight": "1.42em",
}
`;

exports[`typography/elements element typography objects should match snapshots for element typography: labelSmall 1`] = `
{
  "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
  "fontSize": "14px",
  "letterSpacing": "0.16em",
  "lineHeight": "1.42em",
}
`;

exports[`typography/elements fonts objects should match snapshot for all fonts objects 1`] = `
{
  "eyebrow": {
    "className": "umd-eyebrow",
    "color": "#000000",
    "fontSize": "12px",
    "fontWeight": 700,
    "letterSpacing": ".05em",
    "lineHeight": 1,
    "textTransform": "uppercase",
  },
  "labelMedium": {
    "@media (min-width: 1024px)": {
      "fontSize": "16px",
      "letterSpacing": "0.16em",
      "lineHeight": "1.5em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(14px + 0.16vw)",
    },
    "className": "umd-label-sans-medium",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "14px",
    "lineHeight": "1.42em",
  },
  "labelSmall": {
    "className": "umd-label-sans-small",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "14px",
    "letterSpacing": "0.16em",
    "lineHeight": "1.42em",
  },
  "medium": {
    "@media (min-width: 1024px)": {
      "fontSize": "18px",
      "fontWeight": "700",
      "letterSpacing": "-0.01em",
      "lineHeight": "1.11em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(16px + 0.16vw)",
    },
    "className": "umd-interactive-sans-medium",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "16px",
    "lineHeight": "1.125em",
  },
  "small": {
    "className": "umd-interactive-sans-small",
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "16px",
    "fontWeight": "700",
    "letterSpacing": "-0.01em",
    "lineHeight": "1.125em",
  },
}
`;

exports[`typography/elements should match snapshot 1`] = `
{
  "eyebrow": {
    "color": "#000000",
    "fontSize": "12px",
    "fontWeight": 700,
    "letterSpacing": ".05em",
    "lineHeight": 1,
    "textTransform": "uppercase",
  },
  "fonts": {
    "eyebrow": {
      "className": "umd-eyebrow",
      "color": "#000000",
      "fontSize": "12px",
      "fontWeight": 700,
      "letterSpacing": ".05em",
      "lineHeight": 1,
      "textTransform": "uppercase",
    },
    "labelMedium": {
      "@media (min-width: 1024px)": {
        "fontSize": "16px",
        "letterSpacing": "0.16em",
        "lineHeight": "1.5em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(14px + 0.16vw)",
      },
      "className": "umd-label-sans-medium",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "14px",
      "lineHeight": "1.42em",
    },
    "labelSmall": {
      "className": "umd-label-sans-small",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "14px",
      "letterSpacing": "0.16em",
      "lineHeight": "1.42em",
    },
    "medium": {
      "@media (min-width: 1024px)": {
        "fontSize": "18px",
        "fontWeight": "700",
        "letterSpacing": "-0.01em",
        "lineHeight": "1.11em",
      },
      "@media (min-width: 650px)": {
        "fontSize": "calc(16px + 0.16vw)",
      },
      "className": "umd-interactive-sans-medium",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "16px",
      "lineHeight": "1.125em",
    },
    "small": {
      "className": "umd-interactive-sans-small",
      "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
      "fontSize": "16px",
      "fontWeight": "700",
      "letterSpacing": "-0.01em",
      "lineHeight": "1.125em",
    },
  },
  "interativeMedium": {
    "@media (min-width: 1024px)": {
      "fontSize": "18px",
      "fontWeight": "700",
      "letterSpacing": "-0.01em",
      "lineHeight": "1.11em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(16px + 0.16vw)",
    },
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "16px",
    "lineHeight": "1.125em",
  },
  "interativeSmall": {
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "16px",
    "fontWeight": "700",
    "letterSpacing": "-0.01em",
    "lineHeight": "1.125em",
  },
  "labelMedium": {
    "@media (min-width: 1024px)": {
      "fontSize": "16px",
      "letterSpacing": "0.16em",
      "lineHeight": "1.5em",
    },
    "@media (min-width: 650px)": {
      "fontSize": "calc(14px + 0.16vw)",
    },
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "14px",
    "lineHeight": "1.42em",
  },
  "labelSmall": {
    "fontFamily": "'Interstate', Helvetica, Arial, Verdana, sans-serif",
    "fontSize": "14px",
    "letterSpacing": "0.16em",
    "lineHeight": "1.42em",
  },
}
`;
